### ISSUE TITLE
grpc.tools.protoc excludes the present working directory from the default proto_path

### ISSUE THREAD

**Issue by bgilmore** [grpc/grpc#7119]
[grpcio_tools-0.15.0](https://pypi.python.org/pypi/grpcio-tools) pushed today contains a non-functional protoc. The following issues are reproducible using `grpcio_tools-0.15.0-cp27-cp27mu-manylinux1_x86_64.whl`

All generated Python contains the following snippet with an incomplete and syntactically invalid import statement, for example:

```
import
from grpc.beta import implementations as beta_implementations
from grpc.beta import interfaces as beta_interfaces
```

Attempting to import the generated code predictably fails:

```
>>> import test_pb2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "test_pb2.py", line 186
    import
          ^
SyntaxError: invalid syntax
```

Additionally, the --proto_path flag is documented in --help with:

> If not given, the current working directory will be used

But attempting to invoke grpc.tools.protoc without specifying an explicit --proto_path flag returns the following error:
`test.proto: File does not reside within any path specified using --proto_path (or -I)`

**Comment by soltanmm** [comment]
Duplicate of #7101.

**Comment by bgilmore** [comment]
Will the proto_path behavior be fixed by the fixed linked from #7101?

**Comment by soltanmm** [comment]
Is that behavior on the current release (... if so how'd you check that...?), or for a previous release?

**Comment by bgilmore** [comment]
Setting aside the broken codegen in #7101, there's a visible behavior change between 0.14.0 and 0.15.0 that prevents codegen unless you specify --proto_path which _should_ be using the current working directory as a default.

```
~ % source grpc-0.14.0/bin/activate
~ % pip show 'grpcio-tools'
---
Metadata-Version: 2.0
Name: grpcio-tools
Version: 0.14.0
Summary: UNKNOWN
Home-page: UNKNOWN
Author: UNKNOWN
Author-email: UNKNOWN
Installer: pip
License: 3-clause BSD
Location: /home/bgilmore/grpc-0.14.0/lib/python2.7/site-packages
Requires: protobuf
Classifiers:
~ % python -m grpc.tools.protoc test.proto --python_out=. --grpc_python_out=.
[codegen succeeds]
```

```
~ % source grpc-0.15.0/bin/activate
~ % pip show 'grpcio-tools'
---
Metadata-Version: 2.0
Name: grpcio-tools
Version: 0.15.0
Summary: UNKNOWN
Home-page: UNKNOWN
Author: UNKNOWN
Author-email: UNKNOWN
Installer: pip
License: 3-clause BSD
Location: /home/bgilmore/grpc-0.15.0/lib/python2.7/site-packages
Requires: grpcio, protobuf
Classifiers:
~ % python -m grpc.tools.protoc test.proto --python_out=. --grpc_python_out=.
D0628 22:15:57.197173456   22016 ev_posix.c:101]             Using polling engine: poll
test.proto: File does not reside within any path specified using --proto_path (or -I).  You must specify a --proto_path which encompasses this file.  Note that the proto_path must be an exact prefix of the .proto file names -- protoc is too dumb to figure out when two paths (e.g. absolute and relative) are equivalent (it's harder than you think).
~ % python -m grpc.tools.protoc test.proto --python_out=. --grpc_python_out=. --proto_path=.
D0628 22:20:28.452959186   22086 ev_posix.c:101]             Using polling engine: poll
[codegen succeeds, but code is broken per #7101]
```

**Comment by soltanmm** [comment]
Ah, okay, I *think* the problem is that when we started packaging the well-known protos, the underlying `protoc` implementation was made aware of them in locations via user-space flags, which then (for whatever reason) disables automatic inclusion of the present working directory.

Gonna hijack this thread for *just* that issue.

**Comment by soltanmm** [comment]
@kpayson64 @nathanielmanistaatgoogle So, AFAICT this is a really, really, really quick fix (just tack on a `-I.` [or `"-I{}".format(os.getcwd())`, or whatever] in the invocation at the end of `tools/distrib/python/grpcio_tools/grpc/tools/protoc.py`).

I don't know if `protoc` has some sort of ordering deal with multiple include paths and conflicting file names along identical relative paths to the include paths. Also, from the protobuf tutorial:

> In general you should set the --proto_path flag to the root of your project and use fully qualified names for all imports.

... How deep down the rabbit hole do we want to go with this? I can imagine that somewhere in protobuf there's a way to inject the WKTs as something other than a `--proto_path` flag that mucks with the default behavior (but that involves possibly complicating the compilation process and code maintenance). Or that there's a way to `argparse` those flags at the Python-level, munge them a bit, then pass them through to `protoc`.
